client/app/queue/AddCavcDatesModal.jsx

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable no-undefined */
+import React, { useState } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import PropTypes from 'prop-types';
+
+import QueueFlowModal from './components/QueueFlowModal';
+import { requestSave, showErrorMessage } from './uiReducer/uiActions';
+import DateSelector from '../components/DateSelector';
+import TextareaField from '../components/TextareaField';
+import Alert from '../components/Alert';
+import COPY from '../../COPY';
+
+/**
+ * @param {Object} props
+ *  - @param {string}   appealId         The id of the appeal we are creating this cavc remand for. Used to grab
+ *                                       associated decision issues on the appeal from state
+ *  - @param {Object[]} decisionIssues   Issues pulled from state to allow the user to select which are being remanded
+ *  - @param {Object}   error            Error sent from the back end upon submit to be displayed rather than submitting
+ *  - @param {boolean}  highlightInvalid Whether or not to show field validation, set to true upon submit
+ *  - @param {Object}   history          Provided with react router to be able to route to another page upon success
+ */
+
+const AddCavcDatesModal = ({ appealId, error, highlightInvalid, history }) => {
+
+  const [judgementDate, setJudgementDate] = useState(null);
+  const [mandateDate, setMandateDate] = useState(null);
+  const [instructions, setInstructions] = useState(undefined);
+
+  const validJudgementDate = () => Boolean(judgementDate);
+  const validMandateDate = () => Boolean(mandateDate);
+  const validInstructions = () => instructions && instructions.length > 0;
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
```suggestion
  const validInstructions = () => instructions?.length > 0;
```

Reply:
good catch!